ruby/tests/basic.rb

```diff
@@ -784,211 +198,6 @@ def test_map_encode_decode
-      assert m.repeated_msg[0].object_id != m2.repeated_msg[0].object_id
-    end
-
-    def test_eq
-      m = TestMessage.new(:optional_int32 => 42,
-                          :repeated_int32 => [1, 2, 3])
-      m2 = TestMessage.new(:optional_int32 => 43,
-                           :repeated_int32 => [1, 2, 3])
-      assert m != m2
-    end
-
-    def test_enum_lookup
-      assert TestEnum::A == 1
-      assert TestEnum::B == 2
-      assert TestEnum::C == 3
-
-      assert TestEnum::lookup(1) == :A
-      assert TestEnum::lookup(2) == :B
-      assert TestEnum::lookup(3) == :C
-
-      assert TestEnum::resolve(:A) == 1
-      assert TestEnum::resolve(:B) == 2
-      assert TestEnum::resolve(:C) == 3
-    end
-
-    def test_parse_serialize
-      m = TestMessage.new(:optional_int32 => 42,
-                          :optional_string => "hello world",
-                          :optional_enum => :B,
-                          :repeated_string => ["a", "b", "c"],
-                          :repeated_int32 => [42, 43, 44],
-                          :repeated_enum => [:A, :B, :C, 100],
-                          :repeated_msg => [TestMessage2.new(:foo => 1),
-                                            TestMessage2.new(:foo => 2)])
-      data = TestMessage.encode m
-      m2 = TestMessage.decode data
-      assert m == m2
-
-      data = Google::Protobuf.encode m
-      m2 = Google::Protobuf.decode(TestMessage, data)
-      assert m == m2
-    end
-
-    def test_encode_decode_helpers
-      m = TestMessage.new(:optional_string => 'foo', :repeated_string => ['bar1', 'bar2'])
-      assert_equal 'foo', m.optional_string
-      assert_equal ['bar1', 'bar2'], m.repeated_string
-
-      json = m.to_json
-      m2 = TestMessage.decode_json(json)
-      assert_equal 'foo', m2.optional_string
-      assert_equal ['bar1', 'bar2'], m2.repeated_string
-      if RUBY_PLATFORM != "java"
-        assert m2.optional_string.frozen?
-        assert m2.repeated_string[0].frozen?
-      end
-
-      proto = m.to_proto
-      m2 = TestMessage.decode(proto)
-      assert_equal 'foo', m2.optional_string
-      assert_equal ['bar1', 'bar2'], m2.repeated_string
-    end
-
-    def test_protobuf_encode_decode_helpers
-      m = TestMessage.new(:optional_string => 'foo', :repeated_string => ['bar1', 'bar2'])
-      encoded_msg = Google::Protobuf.encode(m)
-      assert_equal m.to_proto, encoded_msg
-
-      decoded_msg = Google::Protobuf.decode(TestMessage, encoded_msg)
-      assert_equal TestMessage.decode(m.to_proto), decoded_msg
-    end
-
-    def test_protobuf_encode_decode_json_helpers
-      m = TestMessage.new(:optional_string => 'foo', :repeated_string => ['bar1', 'bar2'])
-      encoded_msg = Google::Protobuf.encode_json(m)
-      assert_equal m.to_json, encoded_msg
-
-      decoded_msg = Google::Protobuf.decode_json(TestMessage, encoded_msg)
-      assert_equal TestMessage.decode_json(m.to_json), decoded_msg
-    end
-
     def test_to_h
```

Comment:
ditto?

Reply:
`to_h` method for proto2 does not return fields which aren't set. And since the expectations are different between proto2 and proto3, this test remains separate.